src/core/thread/mle.cpp

```diff
@@ -1810,6 +1810,8 @@ ThreadError Mle::HandleParentResponse(const Message &aMessage, const Ip6::Messag
     SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kLinkMargin, sizeof(linkMarginTlv), linkMarginTlv));
     VerifyOrExit(linkMarginTlv.IsValid(), error = kThreadError_Parse);
 
+    VerifyOrExit(threadMessageInfo != NULL, error = kThreadError_NotFound; otLogDebgMle("thread message info not found\n"));
```

Comment:
sounds like a good idea. The assert, however, will end up with a bug that to me, personally, is not much different from segfault, thus may be hard to debug in production, especially when no log messages are left. With the existing VerifyOrExit we were sending some logs.

Anyway, assert sounds good too and I applied the change.